### PR TITLE
feat(openapi-tooling): Update template to use `info.title` as plugin ID

### PR DESCRIPTION
**Pull request by sennyeya**
## Hey, I just made a Pull Request!

While working on a spec in #22205, the specification of pluginId irked me. I think this is a better set up, as it would show up in SwaggerUI as well and gives users a better idea of what they're calling.  

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/afraid-numbers-invite.md
Comment (sennyeya) [inline .changeset/afraid-numbers-invite.md:10]: An alternative approach might be to use the [`--additional-properties=`](https://openapi-generator.tech/docs/configuration) options for the CLI and either autopopulate that from the package context or pull it in from the command line. I think I like that better, more intuitive as a user to just create a client from a given backend. Will rework this.

File: .changeset/long-suns-bow.md
Comment (Rugvip) [inline .changeset/long-suns-bow.md:4]:
```suggestion
'@backstage/plugin-todo-backend': patch
```

File: .changeset/nine-bulldogs-camp.md
Comment (Rugvip) [inline .changeset/nine-bulldogs-camp.md:5]:
```suggestion
'@backstage/plugin-todo-backend': patch
```

File: plugins/catalog-backend/src/schema/openapi.generated.ts
Comment (Rugvip) [inline plugins/catalog-backend/src/schema/openapi.generated.ts:2]:
```suggestion
 * Copyright 2023 The Backstage Authors
```

As long as not all the content is replaced this should stay as the original year, because the rest if just derivative work of that initial copyright.

File: plugins/search-backend/src/schema/openapi.generated.ts
Comment (Rugvip) [inline plugins/search-backend/src/schema/openapi.generated.ts:2]:
```suggestion
 * Copyright 2023 The Backstage Authors
```

File: plugins/todo-backend/src/schema/openapi.generated.ts
Comment (Rugvip) [inline plugins/todo-backend/src/schema/openapi.generated.ts:2]:
```suggestion
 * Copyright 2023 The Backstage Authors
```

**Review by Rugvip — APPROVED** [review]
Agreed! 👍  I don't think `title` is perfect, but it's certainly a nicer pattern than using the `url` imo

File: .changeset/afraid-numbers-invite.md
Comment (Rugvip) [reply .changeset/afraid-numbers-invite.md:10]: There's not a great source of `pluginId` from the package right now though. It's something we'll add in the future for sure though. Let's not derive it from the package name or file structure :p

Will leave it to you whether you want to merge this and tweak it in the future or go for a different approach straight away. I think it could make sense to ship tbh, as a single better state than the current iteration as adoption of this ramps up.

Comment (sennyeya) [reply .changeset/afraid-numbers-invite.md:10]: True, let's iterate on this.

File: plugins/catalog-backend/src/schema/openapi.generated.ts
Comment (sennyeya) [reply plugins/catalog-backend/src/schema/openapi.generated.ts]: There's a prettier issue here where since we regenerate the code everytime, the block comment at the top of the generated code gets evaluated by prettier. Not sure what the best approach is here, same thing on the template side.

Comment (Rugvip) [reply plugins/catalog-backend/src/schema/openapi.generated.ts]: Ah alright, let's not block this PR on that. Could be nice to keep the year in place but that's low prio. Not actually sure what is the correct behavior here is anyway

**Pull request merged**
